library/ssl_tls.c

```diff
@@ -6720,7 +6702,7 @@ int mbedtls_ssl_read( mbedtls_ssl_context *ssl, unsigned char *buf, size_t len )
         if( ssl->in_msgtype == MBEDTLS_SSL_MSG_ALERT )
         {
             MBEDTLS_SSL_DEBUG_MSG( 2, ( "ignoring non-fatal non-closure alert" ) );
-            return( MBEDTLS_ERR_SSL_WANT_READ );
+            continue;
```

Comment:
@RonEld Can you elaborate? We _do_ ignore the message by going into the next iteration here, because the latter will fetch a new record. That's also what's happening when we return `WANT_READ` to the user and have him call `mbedtls_ssl_read` again.